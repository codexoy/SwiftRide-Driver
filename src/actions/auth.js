import {
	ROOT_URL,
	TOKEN_KEY,
	REFRESH_TOKEN_KEY,
	DEFAULT_ERROR,
	signUpRoles
} from "../consts"
import { Linking } from "react-native"
import Storage from "../services/Storage"
import { LOGIN, LOGOUT, API_ERROR } from "../reducers/consts"
import { fetchOrError } from "./utils"

const loginOrRegister = async (endpoint, params, dispatch, callback) => {
	const resp = await dispatch(fetchOrError(endpoint, params))
	if (resp) {
		await setTokens(resp.json.auth_token, resp.json.refresh_token)
		await dispatch(setUser(resp.json.user))
		await callback(resp.json.user)
	} else await callback(undefined)
}

export const directLogin = (email, password, callback) => {
	return async dispatch => {
		const requestParams = {
			method: "POST",
			body: JSON.stringify({ email, password })
		}
		await loginOrRegister(
			"/login/direct",
			requestParams,
			dispatch,
			callback
		)
	}
}

export const register = (params, callback, role = "") => {
	return async dispatch => {
		var data = new FormData()
		Object.keys(params).forEach(key => data.append(key, params[key]))
		const requestParams = {
			method: "POST",
			headers: {
				"Content-Type": "multipart/form-data"
			},
			body: data
		}
		const newCallback = async user => {
			if (user) {
				let roleRequest
				if (role == signUpRoles.student) {
					roleRequest = await dispatch(
						fetchOrError(
							"/user/make_student?teacher_id=" +
								params.teacher_id,
							{
								method: "GET"
							}
						)
					)
				} else if (role == signUpRoles.teacher) {
					roleRequest = await dispatch(
						fetchOrError("/user/make_teacher?", {
							method: "POST",
							body: JSON.stringify({
								price: params.price,
								lesson_duration: params.duration
							})
						})
					)
				}
				if (roleRequest) {
					await callback(user)
					return
				}
			}
			await callback(undefined)
		}

		await loginOrRegister(
			"/login/register",
			requestParams,
			dispatch,
			newCallback
		)
	}
}

const setTokens = async (token, refresh_token) => {
	await Storage.setItem(TOKEN_KEY, token, true)
	await Storage.setItem(REFRESH_TOKEN_KEY, refresh_token, true)
}

export const setUser = user => {
	return dispatch => {
		dispatch({ type: LOGIN, user: user })
	}
}

export const logout = (callback = async () => {}) => {
	return async dispatch => {
		console.log("Logging out")
		await Storage.removeItem(TOKEN_KEY, true)
		await Storage.removeItem(REFRESH_TOKEN_KEY, true)
		await callback()
		setTimeout(() => {
			dispatch({ type: LOGOUT })
		}, 100)
	}
}

export const fetchUser = (callback = () => {}) => {
	/* effectively checks if the token is valid */
	return async dispatch => {
		const resp = await dispatch(
			fetchOrError("/user/me", { method: "GET" }, false)
		)
		if (resp) {
			dispatch(setUser(resp.json.user))
			await callback(resp.json.user)
		} else await callback(undefined)
	}
}

export const exchangeToken = (token, callback) => {
	return async dispatch => {
		try {
			const resp = await dispatch(
				fetchOrError("/login/exchange_token", {
					method: "POST",
					body: JSON.stringify({
						exchange_token: token
					})
				})
			)
			await setTokens(resp.json.auth_token, resp.json.refresh_token)
			await dispatch(fetchUser(callback))
		} catch (error) {
			dispatch({ type: API_ERROR, error: DEFAULT_ERROR })
		}
	}
}

export const openFacebook = (token = "") => {
	let url = ROOT_URL + "/login/facebook"
	if (token) {
		url += "?token=" + token
	}
	Linking.openURL(url)
}
