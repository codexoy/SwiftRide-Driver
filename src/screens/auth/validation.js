import { strings } from "../../i18n"

export const loginValidation = {
	email: {
		email: {
			message: "^" + strings("signin.invalid_email")
		}
	},

	password: {
		length: {
			minimum: 5,
			message: "^" + strings("signin.invalid_password")
		}
	}
}

export const registerValidation = {
	email: {
		email: {
			message: "^" + strings("signin.invalid_email")
		}
	},

	name: {
		length: {
			minimum: 1,
			message: "^" + strings("signin.invalid_field")
		}
	},

	area: {
		length: {
			minimum: 1,
			message: "^" + strings("signin.invalid_field")
		}
	},

	password: {
		length: {
			minimum: 5,
			message: "^" + strings("signin.invalid_email")
		}
	}
}
