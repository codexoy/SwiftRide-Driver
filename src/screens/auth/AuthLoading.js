import React from "react"
import { ActivityIndicator, StyleSheet, StatusBar, View } from "react-native"
import Storage from "../../services/Storage"
import { connect } from "react-redux"
import { TOKEN_KEY } from "../../consts"
import { loadFetchService } from "../../actions/utils"
import { fetchUser, logout } from "../../actions/auth"

export class AuthLoading extends React.Component {
	constructor(props) {
		super(props)
		this._bootstrapAsync()
	}

	// Fetch the token from storage then navigate to our appropriate place
	_bootstrapAsync = async () => {
		this.props.dispatch(loadFetchService())
		await this.props.dispatch(
			fetchUser(async (user = null) => {
				if (user === null) await this.props.dispatch(logout())
				// logging out just to make sure
				this.props.navigation.navigate(user ? "App" : "Auth")
			})
		)
	}

	// Render any loading content that you like here
	render() {
		return (
			<View style={styles.container}>
				<ActivityIndicator />
				<StatusBar barStyle="default" />
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "center"
	}
})

export default connect()(AuthLoading)
