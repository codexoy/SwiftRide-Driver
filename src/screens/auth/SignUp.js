import React, { Fragment } from "react"
import {
	View,
	Text,
	StyleSheet,
	ScrollView,
	KeyboardAvoidingView,
	TouchableOpacity,
	Platform,
	Keyboard
} from "react-native"
import { connect } from "react-redux"
import { register } from "../../actions/auth"
import { API_ERROR, POP_ERROR } from "../../reducers/consts"
import validate, { registerValidation } from "../../actions/validate"
import { strings } from "../../i18n"
import AuthInput from "../../components/AuthInput"
import { MAIN_PADDING, DEFAULT_MESSAGE_TIME, DEFAULT_IMAGE } from "../../consts"
import { Icon } from "react-native-elements"
import LoadingButton from "../../components/LoadingButton"
import SlidingMessage from "../../components/SlidingMessage"
import { popLatestError } from "../../actions/utils"
import UploadProfileImage from "../../components/UploadProfileImage"

export class SignUp extends React.Component {
	constructor(props) {
		super(props)
		this.register = this.register.bind(this)
		this.role = this.props.navigation.getParam("role")
		this.state = {
			api_error: "",
			slidingMessageVisible: false,
			imageError: "",
			image: ""
		}
		this.inputs = {
			email: {},
			name: { iconName: "person", placeholder: strings("signup.name") },
			area: {
				iconName: "person-pin",
				placeholder: strings("signup.area")
			},
			password: { secureTextEntry: true, iconName: "security" }
		}
		Object.keys(this.inputs).forEach(input => {
			this.state[input] = ""
			this.state[input + "Error"] = ""
		})
	}
	componentDidMount() {
		if (Platform.OS === "android") {
			this.keyboardEventListeners = [
				Keyboard.addListener(
					"keyboardDidShow",
					this._handleKeyboardShow
				),
				Keyboard.addListener(
					"keyboardDidHide",
					this._handleKeyboardHide
				)
			]
		}
	}

	componentWillUnmount() {
		this.keyboardEventListeners &&
			this.keyboardEventListeners.forEach(eventListener =>
				eventListener.remove()
			)
	}

	_handleKeyboardHide = () => {
		this.scrollView.scrollTo({ y: 0 })
	}

	_handleKeyboardShow = () => {
		this.scrollView.scrollToEnd()
	}

	componentDidUpdate() {
		const error = this.props.dispatch(popLatestError(API_ERROR))
		if (error) {
			this.setState({
				error,
				slidingMessageVisible: true
			})
		}
	}

	async register() {
		let error,
			errors = []
		Object.keys(this.inputs).forEach(input => {
			error = validate(input, this.state[input], registerValidation)
			if (error) errors.push(error)
			this.setState({ [input + "Error"]: error })
		})
		if (this.state.image == "") {
			this.setState({ imageError: strings("signup.image_required") })
			return
		}

		if (errors.length > 0) return
		this.button.showLoading(true)

		await this.props.dispatch(
			register(
				{
					email: this.state.email,
					area: this.state.area,
					password: this.state.password,
					name: this.state.name,
					image: this.state.image
				},
				user => {
					this.button.showLoading(false)
					if (user) {
						this.setState(
							{ api_error: "", slidingMessageVisible: true },
							() => {
								setTimeout(
									() => this.props.navigation.navigate("App"),
									DEFAULT_MESSAGE_TIME
								)
							}
						)
					}
				}
			)
		)
	}

	renderInputs = () => {
		return Object.keys(this.inputs).map((name, index) => {
			const props = this.inputs[name]
			return (
				<AuthInput
					key={`key${name}`}
					name={name}
					placeholder={props.placeholder || strings("signin." + name)}
					onChangeText={input => this.setState({ [name]: input })}
					value={this.state[name]}
					testID={`r${name}Input`}
					iconName={props.iconName || name}
					errorMessage={this.state[`${name}Error`]}
					validation={registerValidation}
					secureTextEntry={props.secureTextEntry || false}
				/>
			)
		})
	}

	render() {
		return (
			<View style={styles.container}>
				<SlidingMessage
					visible={this.state.slidingMessageVisible}
					error={this.state.api_error}
					success={strings("signup.success")}
					close={() =>
						this.setState({ slidingMessageVisible: false })
					}
				/>
				<TouchableOpacity
					onPress={() => {
						this.props.navigation.goBack()
					}}
					style={styles.backButton}
				>
					<Icon name="arrow-forward" type="material" />
				</TouchableOpacity>
				<KeyboardAvoidingView
					behavior={Platform.OS === "ios" ? "position" : null}
				>
					<ScrollView
						keyboardDismissMode={
							Platform.OS === "ios" ? "interactive" : "on-drag"
						}
						keyboardShouldPersistTaps="handled"
						ref={r => (this.scrollView = r)}
					>
						<View style={styles.formContainer}>
							<UploadProfileImage
								style={styles.profilePic}
								image={this.state.image.uri || DEFAULT_IMAGE}
								upload={async source => {
									this.setState({
										image: source,
										imageError: ""
									})
								}}
							/>
							<Text style={styles.error}>
								{this.state.imageError}
							</Text>
							{this.renderInputs()}
							<LoadingButton
								title={strings("signup.signup_button")}
								onPress={this.register}
								ref={c => (this.button = c)}
								style={styles.button}
								indicatorColor="#fff"
							/>
						</View>
					</ScrollView>
				</KeyboardAvoidingView>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	error: {
		marginTop: 12,
		color: "red",
		fontSize: 12
	},
	backButton: {
		alignSelf: "flex-start",
		marginTop: 12,
		marginLeft: MAIN_PADDING
	},
	form: {
		flex: 1,
		marginTop: MAIN_PADDING
	},
	button: {
		marginTop: 20
	},
	profilePic: {
		width: 80,
		height: 80,
		borderRadius: 40,
		alignSelf: "center"
	},
	container: {
		flex: 1
	},
	formContainer: {
		flex: 1,
		paddingLeft: MAIN_PADDING,
		paddingRight: MAIN_PADDING,
		alignItems: "center",
		paddingBottom: 40
	}
})

const mapStateToProps = state => {
	return {
		errors: state.errors
	}
}

export default connect(mapStateToProps)(SignUp)
