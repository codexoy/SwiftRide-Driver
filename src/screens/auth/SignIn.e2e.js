import React from "react"
import { View, Button, Linking, Text } from "react-native"
import {
	deepLinkingListener,
	deepLinkingRemoveListener
} from "../../actions/utils"
import { connect } from "react-redux"
import {
	exchangeToken,
	openFacebook,
	directLogin,
	setUser
} from "../../actions/auth"
import { Input } from "react-native-elements"
import { API_ERROR, POP_ERROR } from "../../reducers/consts"
import { loginValidation } from "./validation"
import validate from "../../actions/validate"
import { strings } from "../../i18n"
import { popLatestError } from "../../actions/utils"

export class SignIn extends React.Component {
	constructor(props) {
		super(props)
		this.login = this.login.bind(this)
		this.state = {
			email: "",
			password: "",
			error: "",
			emailError: "",
			passwordError: ""
		}
	}

	handleOpenURL = async event => {
		if (event.url) {
			let url = event.url.replace("#_=_", "")
			console.log(`Launched from deeplink ${url}`)
			url = new URLSearchParams(url).toString()
			let regex = /token=(.*)/
			const token = url.match(regex)[1]
			console.log(`New exchange token ${token}`)
			this.props.dispatch(
				exchangeToken(token, user => {
					this.props.navigation.navigate("App")
				})
			)
		}
	}

	async componentDidMount() {
		/* dismiss all errors on focus
		https://stackoverflow.com/questions/49458226/react-native-react-navigation-rerender-panel-on-goback
		*/
		this.willFocusSubscription = this.props.navigation.addListener(
			"willFocus",
			payload => {
				this.setState({
					error: "",
					passwordError: "",
					emailError: ""
				})
			}
		)
		let url = await Linking.getInitialURL()
		console.log("deep linked: " + url)
		await this.handleOpenURL({ url: url })
	}

	async componentWillUnmount() {
		this.willFocusSubscription.remove()
		await deepLinkingRemoveListener(this.handleOpenURL)
	}

	componentDidUpdate() {
		const error = this.props.dispatch(popLatestError(API_ERROR))
		if (error) {
			this.setState({
				error,
				slidingMessageVisible: true
			})
		}
	}

	async login() {
		const emailError = validate("email", this.state.email, loginValidation)
		const passwordError = validate(
			"password",
			this.state.password,
			loginValidation
		)

		this.setState({
			emailError: emailError,
			passwordError: passwordError
		})

		if (emailError || passwordError) return

		if (this.state.email == "t@t.com" && this.state.password == "testing") {
			this.props.dispatch(setUser({ name: "test", teacher_id: 1 }))
			this.props.navigation.navigate("App")
		} else {
			await this.props.dispatch(
				directLogin(this.state.email, this.state.password, () => {
					this.props.navigation.navigate("App")
				})
			)
		}
	}

	render() {
		return (
			<View>
				<Text>Mocked SignIn!</Text>
				<Text testID="error">{this.state.error}</Text>
				<Input
					placeholder={strings("signin.email")}
					onChangeText={email => this.setState({ email })}
					onBlur={() => {
						this.setState({
							emailError: validate(
								"email",
								this.state.email,
								loginValidation
							)
						})
					}}
					value={this.state.email}
					testID="emailInput"
					errorMessage={this.state.emailError}
				/>
				<Input
					placeholder={strings("signin.password")}
					onChangeText={password => this.setState({ password })}
					onBlur={() => {
						this.setState({
							passwordError: validate(
								"password",
								this.state.password,
								loginValidation
							)
						})
					}}
					value={this.state.password}
					secureTextEntry={true}
					testID="passwordInput"
					errorMessage={this.state.passwordError}
				/>
				<Button
					testID="signInButton"
					title={strings("signin.login_button")}
					onPress={this.login}
				/>
				<Button
					title={strings("signin.signup_button")}
					testID="signUpButton"
					onPress={() => {
						this.props.navigation.navigate("SignUp")
					}}
				/>
				<Button
					title={strings("signin.facebook_login")}
					testID="facebookLogin"
					onPress={() => {
						openFacebook()
					}}
				/>
			</View>
		)
	}
}

const mapStateToProps = state => {
	return {
		errors: state.errors
	}
}

export default connect(mapStateToProps)(SignIn)
