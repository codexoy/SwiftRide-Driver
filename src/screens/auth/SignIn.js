import React from "react"
import {
	View,
	ScrollView,
	Text,
	StyleSheet,
	TouchableOpacity,
	KeyboardAvoidingView
} from "react-native"
import {
	deepLinkingListener,
	deepLinkingRemoveListener
} from "../../actions/utils"
import { connect } from "react-redux"
import { exchangeToken, openFacebook, directLogin } from "../../actions/auth"
import { Input, Icon } from "react-native-elements"
import { API_ERROR, POP_ERROR } from "../../reducers/consts"
import { getLatestError } from "../../error_handling"
import { loginValidation } from "./validation"
import validate from "../../actions/validate"
import { strings } from "../../i18n"
import { colors, MAIN_PADDING } from "../../consts"
import Logo from "../../components/Logo"

export class SignIn extends React.Component {
	constructor(props) {
		super(props)
		this.defaultInputColor = "#c9c9c9"
		this.state = {
			email: "",
			password: "",
			error: "",
			emailError: "",
			passwordError: "",
			emailColor: this.defaultInputColor,
			passwordColor: this.defaultInputColor
		}
		this.handleOpenURL = this.handleOpenURL.bind(this)
		this.login = this.login.bind(this)
	}

	handleOpenURL = async event => {
		if (event.url) {
			let url = event.url.replace("#_=_", "")
			console.log(`Launched from deeplink ${url}`)
			url = new URLSearchParams(url).toString()
			let regex = /token=(.*)/
			const token = url.match(regex)[1]
			console.log(`New exchange token ${token}`)
			this.props.dispatch(
				exchangeToken(token, user => {
					this.props.navigation.navigate("App")
				})
			)
		}
	}

	async componentDidMount() {
		/* dismiss all errors on focus
		https://stackoverflow.com/questions/49458226/react-native-react-navigation-rerender-panel-on-goback
		*/
		this.willFocusSubscription = this.props.navigation.addListener(
			"willFocus",
			payload => {
				this.setState({
					error: "",
					passwordError: "",
					emailError: "",
					passwordColor: "",
					emailColor: ""
				})
			}
		)
		deepLinkingListener(this.handleOpenURL)
	}

	async componentWillUnmount() {
		this.willFocusSubscription.remove()
		await deepLinkingRemoveListener(this.handleOpenURL)
	}

	componentDidUpdate() {
		const error = getLatestError(this.props.errors[API_ERROR])
		if (error) {
			this.setState({ error })
			this.props.dispatch({ type: POP_ERROR, errorType: API_ERROR })
		}
	}

	async login() {
		const emailError = validate("email", this.state.email, loginValidation)
		const passwordError = validate(
			"password",
			this.state.password,
			loginValidation
		)

		this.setState({
			emailError: emailError,
			passwordError: passwordError
		})

		if (emailError || passwordError) return

		await this.props.dispatch(
			directLogin(this.state.email, this.state.password, () => {
				this.props.navigation.navigate("App")
			})
		)
	}

	onFocus = param => {
		this.setState({
			[param + "Color"]: colors.blue
		})
	}

	onBlur = param => {
		this.setState({
			[param + "Color"]: this.defaultInputColor,
			[param + "Error"]: validate(
				param,
				this.state[param],
				loginValidation
			)
		})
	}

	render() {
		return (
			<View style={styles.container}>
				<View style={styles.topLogo}>
					<Logo size="medium" />
				</View>
				<KeyboardAvoidingView behavior="padding">
					<ScrollView
						keyboardDismissMode="on-drag"
						keyboardShouldPersistTaps="always"
					>
						<View style={styles.formContainer}>
							<Text testID="error">{this.state.error}</Text>
							<Input
								placeholder={strings("signin.email")}
								onChangeText={email => this.setState({ email })}
								onFocus={() => this.onFocus("email")}
								onBlur={() => this.onBlur("email")}
								value={this.state.email}
								testID="emailInput"
								errorMessage={this.state.emailError}
								inputContainerStyle={styles.inputContainer}
								inputStyle={styles.input}
								leftIcon={
									<View style={styles.inputIcon}>
										<Icon
											name="email"
											type="material"
											size={20}
											color={
												this.state.emailColor ||
												this.defaultInputColor
											}
										/>
									</View>
								}
							/>
							<Input
								placeholder={strings("signin.password")}
								onChangeText={password =>
									this.setState({ password })
								}
								onFocus={() => this.onFocus("password")}
								onBlur={() => this.onBlur("password")}
								value={this.state.password}
								secureTextEntry={true}
								testID="passwordInput"
								errorMessage={this.state.passwordError}
								inputContainerStyle={styles.inputContainer}
								inputStyle={styles.input}
								leftIcon={
									<View style={styles.inputIcon}>
										<Icon
											name="security"
											type="material"
											size={20}
											color={
												this.state.passwordColor ||
												this.defaultInputColor
											}
										/>
									</View>
								}
							/>
							<TouchableOpacity
								testID="signInButton"
								onPress={this.login}
								style={styles.button}
							>
								<Text style={styles.buttonText}>
									{strings("signin.login_button")}
								</Text>
							</TouchableOpacity>
							<Text style={styles.or}>
								{strings("signin.or")}
							</Text>
							<TouchableOpacity
								testID="facebookLogin"
								onPress={openFacebook}
								style={{
									...styles.button,
									...styles.facebook
								}}
							>
								<Text
									style={{
										...styles.buttonText,
										color: "#fff"
									}}
								>
									{strings("signin.facebook_login")}
								</Text>
							</TouchableOpacity>
							<TouchableOpacity
								testID="signUpButton"
								onPress={() => {
									this.props.navigation.navigate("SignUp")
								}}
							>
								<View style={styles.signUpButton}>
									<Text style={styles.callToAction}>
										{strings("signin.not_yet_registered")}
									</Text>
									<Text style={styles.actionButton}>
										{" "}
										{strings("signin.signup_button")}
									</Text>
								</View>
							</TouchableOpacity>
						</View>
					</ScrollView>
				</KeyboardAvoidingView>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	topLogo: {
		flex: 2,
		justifyContent: "center",
		backgroundColor: colors.blue,
		maxHeight: 240
	},
	formContainer: {
		flex: 3,
		paddingLeft: MAIN_PADDING,
		paddingRight: MAIN_PADDING,
		alignItems: "center"
	},
	button: {
		borderRadius: 32,
		padding: 20,
		backgroundColor: "#ececec",
		width: "100%",
		marginTop: 20,
		alignItems: "center"
	},
	buttonText: {
		fontSize: 20,
		color: "#9b9b9b"
	},
	input: {
		padding: 16,
		textAlign: "right"
	},
	inputContainer: {
		borderWidth: 1,
		borderRadius: 28,
		borderColor: "#e0e0e0",
		marginTop: 20
	},
	facebook: {
		backgroundColor: colors.blue,
		marginTop: 0
	},
	or: {
		marginVertical: 16,
		fontWeight: "bold",
		fontSize: 20,
		color: "#c9c9c9"
	},
	callToAction: {
		color: "#9b9b9b"
	},
	signUpButton: {
		flexDirection: "row",
		marginTop: 20
	},
	actionButton: {
		color: colors.blue
	},
	inputIcon: {
		marginLeft: 6
	}
})

const mapStateToProps = state => {
	return {
		errors: state.errors
	}
}

export default connect(mapStateToProps)(SignIn)
