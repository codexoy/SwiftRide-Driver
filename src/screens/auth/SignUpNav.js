import { createStackNavigator } from "react-navigation"
import SignUpAs from "./SignUpAs"
import { SignUp } from "./SignUp"

export default createStackNavigator(
	{ First: SignUpAs, Second: SignUp },
	{
		initialRouteKey: "First",
		headerMode: "none",
		navigationOptions: {
			headerVisible: false
		}
	}
)
