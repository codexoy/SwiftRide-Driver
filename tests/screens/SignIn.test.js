import "react-native"
import React from "react"

import renderer from "react-test-renderer"

import { SignIn } from "../../src/screens/auth/SignIn"

describe("SignIn", () => {
	test("view renders correctly", () => {
		const tree = renderer.create(<SignIn />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
