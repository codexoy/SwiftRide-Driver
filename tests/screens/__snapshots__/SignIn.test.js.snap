// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(12,116,244)",
        "flex": 2,
        "justifyContent": "center",
        "maxHeight": 240,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/logo.png",
          }
        }
        style={
          Object {
            "height": 112,
            "resizeMode": "contain",
            "width": 140,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 40,
          }
        }
      >
        Dryvo
      </Text>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <RCTScrollView
      keyboardDismissMode="interactive"
      keyboardShouldPersistTaps="handled"
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 3,
              "paddingLeft": 26,
              "paddingRight": 26,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#e0e0e0",
                  "borderRadius": 28,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginTop": 12,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 15,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginLeft": 6,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                      theme={
                        Object {
                          "colors": Object {
                            "disabled": "hsl(208, 8%, 90%)",
                            "divider": "#bcbbc1",
                            "error": "#ff190c",
                            "grey0": "#393e42",
                            "grey1": "#43484d",
                            "grey2": "#5e6977",
                            "grey3": "#86939e",
                            "grey4": "#bdc6cf",
                            "grey5": "#e1e8ee",
                            "greyOutline": "#bbb",
                            "platform": Object {
                              "android": Object {
                                "error": "#f44336",
                                "primary": "#2196f3",
                                "secondary": "#9C27B0",
                                "success": "#4caf50",
                                "warning": "#ffeb3b",
                              },
                              "ios": Object {
                                "error": "#ff3b30",
                                "primary": "#007aff",
                                "secondary": "#5856d6",
                                "success": "#4cd964",
                                "warning": "#ffcc00",
                              },
                            },
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#8F0CE8",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        }
                      }
                      underlayColor="white"
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": "#c9c9c9",
                              "fontSize": 20,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <TextInput
                allowFontScaling={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="אימייל"
                secureTextEntry={false}
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "black",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                    "padding": 16,
                    "textAlign": "right",
                  }
                }
                testID="emailInput"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#e0e0e0",
                  "borderRadius": 28,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginTop": 12,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 15,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginLeft": 6,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                      theme={
                        Object {
                          "colors": Object {
                            "disabled": "hsl(208, 8%, 90%)",
                            "divider": "#bcbbc1",
                            "error": "#ff190c",
                            "grey0": "#393e42",
                            "grey1": "#43484d",
                            "grey2": "#5e6977",
                            "grey3": "#86939e",
                            "grey4": "#bdc6cf",
                            "grey5": "#e1e8ee",
                            "greyOutline": "#bbb",
                            "platform": Object {
                              "android": Object {
                                "error": "#f44336",
                                "primary": "#2196f3",
                                "secondary": "#9C27B0",
                                "success": "#4caf50",
                                "warning": "#ffeb3b",
                              },
                              "ios": Object {
                                "error": "#ff3b30",
                                "primary": "#007aff",
                                "secondary": "#5856d6",
                                "success": "#4cd964",
                                "warning": "#ffcc00",
                              },
                            },
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#8F0CE8",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        }
                      }
                      underlayColor="white"
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": "#c9c9c9",
                              "fontSize": 20,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <TextInput
                allowFontScaling={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="סיסמה"
                secureTextEntry={true}
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "black",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                    "padding": 16,
                    "textAlign": "right",
                  }
                }
                testID="passwordInput"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ececec",
                  "borderRadius": 32,
                  "height": 64,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#9b9b9b",
                    "fontSize": 20,
                  }
                }
              >
                התחברות
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#c9c9c9",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginVertical": 16,
              }
            }
          >
            או
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgb(12,116,244)",
                  "borderRadius": 32,
                  "height": 64,
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 20,
                  }
                }
              >
                התחבר עם Facebook
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="signUpButton"
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginTop": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#9b9b9b",
                  }
                }
              >
                עוד אין לך חשבון?
              </Text>
              <Text
                style={
                  Object {
                    "color": "rgb(12,116,244)",
                  }
                }
              >
                 
                הרשמה
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
